src/test/lib/statistics/cardinality_estimator_test.cpp

```diff
@@ -913,4 +914,138 @@ TEST_F(CardinalityEstimatorTest, NonQueryNodes) {
   EXPECT_EQ(estimator.estimate_cardinality(DummyTableNode::make()), 0.0f);
 }
 
+// Usually, we assume that we know nothing about predicates getting their values from uncorrelated subqueries and we
+// simply forward the input estimations. However, special predicates with subqueries can stem from join rewrites (e.g.,
+// JoinToPredicateRewriteRule). We estimate the resulting plans just like we would have done it for the original semi-
+// joins. The following two test cases ensure the correct behavior for the possible rewrites.
+TEST_F(CardinalityEstimatorTest, ValueScanWithUncorrelatedSubquery) {
+  // Case (i): Predicate column = <subquery>
+  // clang-format off
+  const auto subquery =
+  ProjectionNode::make(expression_vector(c_y),
+    PredicateNode::make(equals_(c_y, value_(2)),
+      node_c));
+  // clang-format on
+
+  auto lqp = PredicateNode::make(equals_(a_a, lqp_subquery_(subquery)), node_a);
```

Comment:
Do we allow such queries and only throw when the subquery returns too many rows?